### PR TITLE
Patch deserialization bug in 2.13.1

### PR DESCRIPTION
**Pull request by davidmrdavid**
As in title. Also adds de-/serialization tests in the smoke tests thanks to @bachuv

### CODE REVIEW

**Review by cgillum — APPROVED** [review]
Assuming we've done the manual validation of the fix and things are working as expected, LGTM.

File: test/SmokeTests/OOProcSmokeTests/durableJS/DurableFunctionsOrchestratorJS/index.js [hunk 1]
```diff
@@ -13,12 +13,15 @@ const df = require("durable-functions");
 
 module.exports = df.orchestrator(function* (context) {
     const outputs = [];
+    const city = {city:"Paris", country:"France"};
 
     // Replace "Hello" with the name of your Durable Activity Function.
     outputs.push(yield context.df.callActivity("Hello", "Tokyo"));
     outputs.push(yield context.df.callActivity("Hello", "Seattle"));
     outputs.push(yield context.df.callActivity("Hello", "London"));
+    outputs.push(yield context.df.callActivity("Hello", "123"));
```
Comment (davidmrdavid) [inline test/SmokeTests/OOProcSmokeTests/durableJS/DurableFunctionsOrchestratorJS/index.js:22]: nit: can we add a case with numbers as input?

**Review by davidmrdavid — COMMENTED** [review]
LGTM as well. Just left a tiny nitpick for the JS case

File: test/SmokeTests/OOProcSmokeTests/durableJS/DurableFunctionsOrchestratorJS/index.js (hunk 1 above)
Comment (davidmrdavid) [reply test/SmokeTests/OOProcSmokeTests/durableJS/DurableFunctionsOrchestratorJS/index.js:22]: This is testing a string with numbers, which is a good edgfe case, but one with "real" ints would be good

Comment (bachuv) [reply test/SmokeTests/OOProcSmokeTests/durableJS/DurableFunctionsOrchestratorJS/index.js]: Yes, I didn't mean to make it a string. I updated it to `123`.

**Pull request merged**